File: WebApi/TikTakWebAPI/models/VideoModel.cs
using System.Text.Json.Serialization;

namespace TikTakWebAPI.Models
{
    public class VideoModel
    {
        [JsonPropertyName("name")]
        public string? Name { get; set; }

        [JsonPropertyName("serverPath")]
        public string? ServerPath { get; set; }

        public VideoModel()
        { }
        public VideoModel(string name, string serverPath)
        {
            Name = name;
            ServerPath = serverPath;
        }
    }
}

File: tik_tak_app/lib/models/video_model.dart
class VideoModel {
  final String name;
  final String serverPath;
  String indexUri;

  VideoModel({required this.name, required this.serverPath, required this.indexUri});

  factory VideoModel.fromJson(Map<String, dynamic> json) {
    return VideoModel(name: json['name'], serverPath: json['serverPath'], indexUri: '');
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};

    data['name'] = name;
    data['serverPath'] = serverPath;
    data['indexUri'] = indexUri;

    return data;
  }
}
